ComfortZoneApp/app/src/main/java/com/example/comfortzone/Utils/LocationUtil.java

```diff
@@ -42,15 +42,18 @@ public static void getLastLocation(Activity activity, Context context, FusedLoca
                     @Override
                     public void onComplete(@NonNull Task<Location> task) {
                         Location location = task.getResult();
-                        long locationTime = location.getTime();
-                        long currentTime = System.currentTimeMillis();
-                        long diff = currentTime - locationTime;
-
-                        if (location == null || diff > ONE_HOUR_MILLI) {
```

Comment:
i guess doing `if (location == null || System.currentTimeMillis() - location.getTime() > ONE_HOUR_MILLI) {` should work? so you don't need to repeat the call in the else case.. either way is fine :)